### PR TITLE
Remove volumeName from disks

### LINKED ISSUE #1024: Remove volumeName and reference volumes from disks with the disk name
###  Here what we do right now with disks

We reference a volume with an extra `volumeName` entry.

Here an example disk entry:

```yaml
      - disk:
          bus: virtio
        name: registrydisk
        volumeName: registryvolume
```

and here the matching volumes entry:
 
```yaml
  volumes:
  - name: registryvolume
    registryDisk:
      image: kubevirt/cirros-registry-disk-demo:devel
```

On pods it works differently. There every `volumeMount` has a name, which also matches the name of a volume.

We can do the same. Let the name of the disk reference a volume and remove `volumeName` in order to converge with how things are expressed on pods.

Follow up of this discussion: https://groups.google.com/d/msgid/kubevirt-dev/cfced0d7-7e24-4e99-a837-e1f2386ae94d%40googlegroups.com?utm_medium=email&utm_source=footer

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Disks and Volumes now need to have matching names, similar to Interfaces
and Networks. This makes the API more consistent.

Before:

```yaml
    spec:
      domain:
        devices:
          disks:
          - disk:
              bus: virtio
            name: registrydisk
            volumeName: registryvolume
      volumes:
      - name: registryvolume
        registryDisk:
          image: registry:5000/kubevirt/cirros-registry-disk-demo:devel
```

After:

```yaml
    spec:
      domain:
        devices:
          disks:
          - disk:
              bus: virtio
            name: registrydisk
      volumes:
      - name: registrydisk
        registryDisk:
          image: registry:5000/kubevirt/cirros-registry-disk-demo:devel
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1024

**Special notes for your reviewer**:

See also the mailing list discussion: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/kubevirt-dev/17EW5fKst4M/oD9EId1bBQAJ.

**Release note**:

```release-note
API incompatible change: "volumeName" got removed. The name of a disk and the name of a volume must match. API version is bumped to `v1alpha3`.
```

### CODE REVIEW

**Comment by rmohr** [comment]
This is the last incompatible API change which I know of. We should probably come up with a roadmap on when we plan to move our API to `v1beta1`, which means that we expect no incompatible API changes anymore until we reach `v1`.

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -55,10 +55,10 @@ const SubresourceGroupName = "subresources.kubevirt.io"
 const DefaultGracePeriodSeconds int64 = 30
 
 // GroupVersion is group version used to register these objects
-var GroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1alpha2"}
+var GroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1alpha3"}
```
Comment (cynepco3hahue) [inline pkg/api/v1/types.go]: Just for curiosity, when do we decide to increase API version?

**Comment by cynepco3hahue** [comment]
I think we will need to update docs as well.

File: pkg/api/v1/types.go (hunk 1 above)
Comment (rmohr) [reply pkg/api/v1/types.go]: I increase it when it is incommpatible with previous VMI definitions. `v1alpha2` definitions will not work anymore after this change.

**Comment by rmohr** [comment]
This should be considered for merging after the next release.

/hold

**Comment by fabiand** [comment]
Yes, and it looks like there are additional API changes which we want to do, putting them into one release will probably be helpful to users.

@rmohr we could also look at the feature in CRDs which allows to migrate from one scheme to anohter to ease user updates.

**Comment by fabiand** [comment]
xref #1576

**Comment by davidvossel** [comment]
I'm in favor of this change

The functional tests are complaining about some validation webhook issues. Looks like it could be related to this.

**Comment by rmohr** [comment]
test

**Comment by davidvossel** [comment]
travis failure :(

**Comment by rmohr** [comment]
Some of them seem to be related. Probably new changes from rebasing: 


```
Summarizing 4 Failures:

[Fail] Storage Starting a VirtualMachineInstance With ephemeral alpine PVC [It] should not persist data 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:1381

[Fail] Config With a ConfigMap defined With a single volume [It] Should be the fs layout the same for a pod and vmi 
/root/go/src/kubevirt.io/kubevirt/tests/config_test.go:90

[Fail] Config With a Secret defined With a single volume [It] Should be the fs layout the same for a pod and vmi 
/root/go/src/kubevirt.io/kubevirt/tests/config_test.go:185

[Fail] VirtualMachineInstanceReplicaSet [It] should reject POST if validation webhoook deems the spec is invalid 
/root/go/src/kubevirt.io/kubevirt/tests/replicaset_test.go:154

Ran 171 of 201 Specs in 4680.941 seconds
```

**Comment by fabiand** [comment]
Ping?

**Comment by davidvossel** [comment]
we can move forward with this whenever we decide it's okay to increment the version field again.

**Comment by stu-gott** [comment]
@davidvossel, Ah. thanks for the rebase!

**Comment by davidvossel** [comment]
The related user-guide documentation PR can be found here. https://github.com/kubevirt/user-guide/pull/192

**Comment by davidvossel** [comment]
@stu-gott Would you be able to give another review to verify the couple of changes I made. I think this should be good to go now.

**Comment by davidvossel** [comment]
The functional tests revealed a real regression.  I need to finish up investigating this before we can merge.

**Comment by davidvossel** [comment]
functional tests should look better this run. We'll wait for CI again before merging.

**Pull request merged**